### PR TITLE
Add performance.md and add instruction for linking tcmalloc

### PR DESCRIPTION
**Pull request by BSBandme**

### CODE REVIEW

File: benchmarks/README.md [hunk 1]
```diff
@@ -143,3 +154,7 @@ We would like to add more data sets.  In general we will favor data sets
 that make the overall suite diverse without being too large or having
 too many similar tests.  Ideally everyone can run through the entire
 suite without the test run getting too long.
+
+## Sample result
+
+A Sample benchmark result can be seen [here](https://github.com/google/protobuf/blob/master/docs/Performance.md)
```
Comment (xfxyjwf) [inline benchmarks/README.md]: Move this to the first paragraph. I.e.:

This directory contains ... test a variety of performance scenarios against your protobuf language runtime. **If you are looking for performance numbers of officially support languages, see [here]**.

File: docs/Performance.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+<th colspan="2">deserializeFromByteArray</th>
+<th colspan="2">deserializeFromByteString</th>
+<th colspan="2">deserializeFromMemoryStream</th>
+<th colspan="2">cpp-generated-code</th>
+<th colspan="2">cpp-reflection</th>
+<th colspan="2">pure-python</th>
+</tr>
+<tr>
+<td>google_message1_proto2</td>
+<td>589ns</td><td>368.717MB/s</td>
```
Comment (xfxyjwf) [inline docs/Performance.md]: Remove the "xxx ns" column. That number isn't useful.

File: docs/Performance.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+<th colspan="2">reuse</th>
+<th colspan="2">deserializeFromByteArray</th>
+<th colspan="2">deserializeFromByteString</th>
+<th colspan="2">deserializeFromMemoryStream</th>
+<th colspan="2">cpp-generated-code</th>
+<th colspan="2">cpp-reflection</th>
+<th colspan="2">pure-python</th>
+</tr>
+<tr>
+<td>google_message1_proto2</td>
```
Comment (xfxyjwf) [inline docs/Performance.md]: Can you include the result with tcmalloc for C++?

File: docs/Performance.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+## Parsing performance
+<table>
+<tbody><tr>
+<th rowspan="2"> </th>
+<th colspan="6" rowspan="1">cpp</th>
+<th colspan="6" rowspan="1">java</th>
+<th colspan="6" rowspan="1">python</th>
+</tr>
+<tr>
+<th colspan="2">new</th>
```
Comment (xfxyjwf) [inline docs/Performance.md]: Add a section to describe these "new","newarena","reuse".

File: benchmarks/README.md (hunk 1 above)
Comment (BSBandme) [reply benchmarks/README.md]: Done

File: docs/Performance.md (hunk 2 above)
Comment (BSBandme) [reply docs/Performance.md]: Done

File: docs/Performance.md (hunk 4 above)
Comment (BSBandme) [reply docs/Performance.md]: Done by add some explanation in column title.

File: docs/Performance.md (hunk 3 above)
Comment (BSBandme) [reply docs/Performance.md]: ack.

File: benchmarks/README.md [hunk 5]
```diff
@@ -3,7 +3,9 @@
 
 This directory contains benchmarking schemas and data sets that you
 can use to test a variety of performance scenarios against your
-protobuf language runtime.
+protobuf language runtime. If you are looking for performance 
+numbers of officially support languages, see [here](
+		https://github.com/google/protobuf/blob/master/docs/Performance.md)
```
Comment (xfxyjwf) [inline benchmarks/README.md]: Remove the leading tabs.

File: docs/Performance.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+## Parsing performance
+<table>
+<tbody><tr>
+<th rowspan="2"> </th>
+<th colspan="3" rowspan="1">cpp</th>
+<th colspan="3" rowspan="1">java</th>
+<th colspan="3" rowspan="1">python</th>
+</tr>
+<tr>
+<th colspan="1">new(use new operator for creating message instance)</th>
```
Comment (xfxyjwf) [inline docs/Performance.md]: This isn't very readable when embedded into the table column. Can you add it as a separate list before the table? I should cover C++ and also other languages.

File: docs/Performance.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+<th colspan="1">newarena(using arena for creating new message instance)</th>
+<th colspan="1">reuse(reuse the same message instance for parsing)</th>
+<th colspan="1">deserializeFromByteArray</th>
```
Comment (xfxyjwf) [inline docs/Performance.md]: use simpler names for java? For example, byte[], ByteString, InputStream.

Comment (BSBandme) [reply docs/Performance.md]: Done

File: docs/Performance.md (hunk 6 above)
Comment (BSBandme) [reply docs/Performance.md]: Done

File: benchmarks/README.md (hunk 5 above)
Comment (BSBandme) [reply benchmarks/README.md]: Done

**Comment by BSBandme** [comment]
tcmalloc results added

File: docs/Performance.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
+# Protobuf Perforamcne
+This benchmark result is tested on workstation with processor of Intel® Xeon® Processor E5-2630 and 32GB RAM
+
+This table contains 3 languages' results:
+
+* **CPP** - For CPP there're 3 kinds of parsing ways:
```
Comment (xfxyjwf) [inline docs/Performance.md]: CPP => C++

File: docs/Performance.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
+# Protobuf Perforamcne
```
Comment (xfxyjwf) [inline docs/Performance.md]: Change the name of this file to doc/performance.md (lower-case p).

File: docs/Performance.md (hunk 8 above)
Comment (BSBandme) [reply docs/Performance.md]: Done

File: docs/Performance.md (hunk 9 above)
Comment (BSBandme) [reply docs/Performance.md]: Done

**Pull request merged**